gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -464,6 +525,159 @@ public License getLicense(LicenseId license, LicenseOption... options) {
     }
   }
 
+  @Override
+  public Operation get(final OperationId operationId, OperationOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Operation answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Operation>() {
+            @Override
+            public com.google.api.services.compute.model.Operation call() {
+              if (operationId instanceof RegionOperationId) {
```

Comment:
replace to switch when adding `type` to operationId. Also, consider making `OperationId` a base class instead of an interface with a package scope constructor to avoid getting "unrecognized" instances.